spec/pactas_itero/client_spec.rb

```diff
@@ -244,6 +264,13 @@
   end
 
   context "error handling" do
+    it "raises on 400" do
```

Comment:
I'm not sure if the 400 the right error code is, but I would to response with a 401 headers key missing

Reply:
The Billwerk API returns 400 if legal entity is not set (and 403 if the id is wrong). I just added this test case because the error mapping for 400 exists already but was missing in the tests. (I just noticed, same for 403, so I added one: https://github.com/shipcloud/billwerk/pull/152/commits/f7946ef22d2c32f3a53c88c9f149c639ab752663)

Instead of making the request and failing on it if legal enity id is missing, we could also validate the presence of legal entity id before we make the actual request and raise some custom exception like `LegalEntityIdMissing`. WDYT?

Reply:
Thanks for the changes, but I'm not sure that we can make the Legal Entity mandatory because the Gem is public, for us is needed because of our Billwerk plan, but someone else might not need to send the LegalEntity to Billwerk

Reply:
Yes fair point, then I guess it has to stay optional. So is it fine as is or should I change something?